### PR TITLE
uplift test coverage

### PR DESCRIPTION
**Pull request by aminalaee**
Uplift test coverage and enforce 100% coverage in the pipeline.

### CODE REVIEW

File: dashboard/datasource.py [hunk 1]
```diff
@@ -37,39 +39,39 @@ def func():
 
 class DataSource:
     def search(self, search_term: str) -> "DataSource":
-        raise NotImplementedError()
+        raise NotImplementedError()  # pragma: no cover
```
Comment (Kludex) [inline dashboard/datasource.py:42]: What's the issue here?

Comment (aminalaee) [reply dashboard/datasource.py:42]: There's no issue I think.
This is an interface which doesn't implement these methods, so we don't have test coverage here.
Child classes like MockDataSource implement that and have coverage but the interface doesn't.

Comment (Kludex) [reply dashboard/datasource.py:42]: I was sleepy. I thought it was `# noqa` :sweat_smile: 

Thanks!

Comment (aminalaee) [reply dashboard/datasource.py:42]: No worries 😁
I'll bump coverage to 100. Now it's 99.

**Pull request merged**